Add extradimensional orange, holymelon, meatwheat, worldpeas mutations

slarticodefast opened the pull request:
## About the PR
Adds four classic plant mutations from SS13 to the game (see the [tg station wiki](https://tgstation13.org/wiki/Guide_to_hydroponics) for reference):

- extradimensional orange, mutated from oranges
- holymelon, mutated from watermelons
- meatwheat, mutated from wheat
- worldpeas, mutated from laughin' peas

Draft for now until my other PR adding haloperidol for the extradimensional orange is merged (see #27454).

## Why / Balance
The extradimensional orange serves as an alternative source of haloperidol (the other being chemistry). The same as in SS13.

The holymelon is the same as the normal melon, except it contains holy water instead of water. Is serves as an alternative source of holy water, when there is no chaplain or bible available. For now it has no anti-magic properties, but these could be implemented once the magic rework is finished.

The watermelon sprite has been changed to the tg version so it looks more similar to the holymelon.

Meatwheat gets squished into meatwheat clumps on use in hand. These were already in the game, but only obtainable through salvage. Serves as an alternative "vegetarian" source of food for the chef. Cooking recipes will have to be added in a separate PR.

World peas are a source of pax and happiness. Since these are dangerous reagents this has been balanced by very long growth time for the plant and low ingredient amounts. Also it is a two step mutation process similar to killer tomatos.

## Technical details
sprites and YAML only

## Media

https://github.com/space-wizards/space-station-14/assets/161409025/5632c9ac-7091-4950-aacd-ae6806d82f2b

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
none

**Changelog**

:cl:
- add: Extradimensional orange, holymelon, meatwheat and world peas plant mutations have been added. Obtain them by mutating oranges, watermelons, wheat and laughin' peas respectively.

lzk228 commented on Resources/Locale/en-US/seeds/seeds.ftl:
```diff
@@ -110,8 +116,10 @@ seeds-berries-name = berries
+seeds-pea-name = peas
 seeds-pea-display-name = pea vines
+seeds-worldpea-name = world peas
+seeds-worldpea-display-name = world peas
```
```suggestion
seeds-worldpea-display-name = world pea vines
```

slarticodefast replied:
done

notquitehadouken commented:
orange

roTAte

keronshb commented:
> The holymelon is the same as the normal melon, except it contains holy water instead of water. Is serves as an alternative source of holy water, when there is no chaplain or bible available. For now it has no anti-magic properties, but these could be implemented once the magic rework is finished.

We're not going to add antimagic properties as they were too much of a hard counter in SS13.

slarticodefast commented:
> We're not going to add antimagic properties as they were too much of a hard counter in SS13.

I don't know much about SS13 to be honest. For now I think it is worth it to include the holymelon even without anti magic. Maybe someone finds a way to add some additional effect later without it being a hard counter.

keronshb commented:
> > We're not going to add antimagic properties as they were too much of a hard counter in SS13.
> 
> I don't know much about SS13 to be honest. For now I think it is worth it to include the holymelon even without anti magic. Maybe someone finds a way to add some additional effect later without it being a hard counter.

Adding the holymelon itself is probably fine, just no antimagic properties for it.

Emisse commented:
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

slarticodefast commented:
Please reopen this and set it to draft.
It requires my other PR to be merged first.

ps3moira commented:
please reopen! this PR is awesome

slarticodefast commented:
My other PR introducing haloperidol was merged, so I can finally undraft this. Ready for review.

slarticodefast commented:
> maybe them having the reagents that causes the meatification process?

The problem with that is that produce is just a solution container, so the meatification reaction would happen instantly inside the meatwheat clump. And disabling reactions inside would be really bad because they could then be used as a eatable cryobeaker.

slarticodefast commented:
That would work, but I think I will keep it simple for now and add recipes in another PR. Having full meat come out of the grinder seems a little weird to me. For now the meatwheat can be used for proteins and can be eaten by lizards. I will try to make recipes accept one from several ingredient options later.

Iamgoofball commented:
>We're not going to add antimagic properties as they were too much of a hard counter in SS13.

This is silly and off the cuff, especially since you neglected to mention that the SS13 version is limited usage with no recharging on melons, it's not just permanent 24/7 anti-magic, you can absolutely punch through it

slarticodefast commented:
Added FoodSequence support to the new produce. This also means meatwheat can be used for burgers.
Ed also implemented the sprite rescaling for the food sequence, so this PR will have to wait for his version.

Everturning commented:
YES!!!!!
